--- origsrc/libgda-5.2.2/configure.ac	2013-12-23 06:39:56.000000000 -0600
+++ src/libgda-5.2.2/configure.ac	2015-01-08 11:41:04.330028200 -0600
@@ -342,14 +342,16 @@ EXPORT_PROV_SYM_REGEX='-export-symbols-r
 EXPORT_UI_SYM_REGEX='-export-symbols-regex "^(gdaui_).*"'
 AC_MSG_CHECKING([for platform])
 platform_win32=no
+platform_cygwin=no
 platform_carbon=no
 have_ige=no
 linklibext=".so"
 case "$host" in
 *-cygwin*)
     AC_MSG_RESULT([Win32 - cygwin])
+    platform_cygwin=yes
     NO_UNDEFINED='-no-undefined'
-    LIBTOOL_EXPORT_OPTIONS=$EXPORT_SYM_REGEX
+    LIBTOOL_EXPORT_OPTIONS=
     LIBTOOL_PROV_EXPORT_OPTIONS=$EXPORT_PROV_SYM_REGEX
     LIBTOOL_UI_EXPORT_OPTIONS=$EXPORT_UI_SYM_REGEX
     linklibext=".dll.a"
@@ -402,6 +404,7 @@ case "$host" in
 esac
 
 AM_CONDITIONAL(PLATFORM_WIN32, [test $platform_win32 = yes])
+AM_CONDITIONAL(PLATFORM_CYGWIN, [test $platform_win32 = yes])
 AM_CONDITIONAL(PLATFORM_CARBON, [test $platform_carbon = yes])
 AM_CONDITIONAL(HAVE_IGE, [test $have_ige = yes])
 AC_SUBST(LIBTOOL_EXPORT_OPTIONS)
@@ -688,7 +691,7 @@ dnl test for system-installed SQLite
 dnl If no SQLite is found or if the SQLite found does not have the sqlite3_table_column_metadata()
 dnl function call, then use the embedded version
 dnl also, on Win32, always use the embedded version
-if test x"$platform_win32" = "xyes"
+if test x"$platform_win32" = "xyes" -o x"$platform_cygwin" = "xyes"
 then
         AM_CONDITIONAL(HAVE_SQLITE, false)
 else
